dao/mongo.go

```diff
@@ -179,124 +232,97 @@ func (m *MongoService) CreatePractitionerResourceForInsolvencyCase(dao *models.P
 		log.Error(err)
-		return http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction %s", transactionID)
+		return nil, http.StatusBadRequest, err
 	}
 
-	return http.StatusCreated, nil
+	return practitionersData, http.StatusCreated, nil
 }
 
-// GetPractitionerResources gets a list of all practitioners for an insolvency case with the specified transactionID
-func (m *MongoService) GetPractitionerResources(transactionID string) ([]models.PractitionerResourceDao, error) {
-	var insolvencyResource models.InsolvencyResourceDao
+// GetInsolvencyPractitionerByTransactionID gets a list of all practitioners for an insolvency case with the specified transactionID
+func (m *MongoService) GetInsolvencyPractitionerByTransactionID(transactionID string) (string, error) {
+	var insolvencyResource models.InsolvencyResourceDto
 	collection := m.db.Collection(m.CollectionName)
 
 	filter := bson.M{"transaction_id": transactionID}
 
 	// Retrieve insolvency case from Mongo
-	opts := options.FindOne().SetProjection(bson.M{"_id": 0, "data.practitioners": 1})
-	storedPractitioners := collection.FindOne(context.Background(), filter, opts)
+	storedPractitioners := collection.FindOne(context.Background(), filter)
+
 	err := storedPractitioners.Err()
 	if err != nil {
 		if err == mongo.ErrNoDocuments {
 			log.Debug("no insolvency case found for transaction id", log.Data{"transaction_id": transactionID})
-			return nil, nil
+			return "", nil
 		}
 		log.Error(err)
-		return nil, err
+		return "", err
 	}
 
 	err = storedPractitioners.Decode(&insolvencyResource)
 	if err != nil {
 		log.Error(err)
-		return nil, err
+		return "", err
 	}
 
 	// Return an empty array instead of nil so the handler can check
 	// that there are no practitioners
-	if insolvencyResource.Data.Practitioners == nil {
-		return make([]models.PractitionerResourceDao, 0), nil
+	if len(insolvencyResource.Data.Practitioners) == 0 {
+		return "", nil
 	}
 
 	return insolvencyResource.Data.Practitioners, nil
 }
 
-// GetPractitionerResource gets a single practitioner for an insolvency case with the specified transactionID and practitionerID
-func (m *MongoService) GetPractitionerResource(practitionerID string, transactionID string) (models.PractitionerResourceDao, error) {
-
-	var insolvencyResource models.InsolvencyResourceDao
-	collection := m.db.Collection(m.CollectionName)
+// GetPractitionersByIds gets a single practitioner for an insolvency case with the specified transactionID and practitionerID
+func (m *MongoService) GetPractitionersByIds(practitionerIDs []string, transactionID string) ([]models.PractitionerResourceDto, error) {
+	var insolvencyResource []models.PractitionerResourceDto
+	collection := m.db.Collection(PractitionerCollectionName)
 
 	filter := bson.M{
-		"transaction_id":        transactionID,
-		"data.practitioners.id": practitionerID,
+		"data.id": bson.M{
+			"$in": bson.A{practitionerIDs},
+		},
```

Comment:
Shouldn't this request get practitioner specific to the transaction?

Reply:
discussed